### ISSUE TITLE
More than 10x slowdown with `1.0.0rc0`.

### ISSUE THREAD

**Issue by s-banach** [plotly/Kaleido#231]
Simple test script:

```python
from time import perf_counter

import plotly.express as px

fig = px.scatter(x=[1, 2, 3], y=[1, 2, 3])

start = perf_counter()
for _ in range(10):
    fig.to_image(format="svg", engine="kaleido")
end = perf_counter()
print(f"{end - start:.02f}")
```

Time with `0.1.0.post1`:  `0.95 seconds`
Time with `1.0.0rc0`: `10.78 seconds`

**Comment by s-banach** [comment]
I'm assuming it does something like open a new copy of chrome for each image.
In that case, please include guidance on how to make a persistent instance of chrome to generate multiple images.
Even better, a pool of chrome instances that can be used to render a list of figures.

**Comment by ayjayt** [comment]
Yeah, that's the issue. I want to upgrade the API soon and deprecate this one.

@s-banach, are you comfortable using `asyncio` for the speedup?

**Comment by s-banach** [comment]
Yes!

**Comment by ayjayt** [comment]
Alright, give me an hour or two and I"ll write and test a quickstart- it should make its way in to the README as well....

**Comment by ayjayt** [comment]
Hey, @s-banach, some news:

Basically the `v1.0.0rc0` is the old API for backwards comparability with the new engine-> not great.
`kaleido2` will be new api + new engine, I'm going to do it this weekend.

That said, I pushed a branch (https://github.com/plotly/Kaleido/tree/andrew/factor_for_faster) which allows you to run code like this:
```python
  from choreographer import 
  import kaleido
  async with Browser() as browser:
    spec = scope.make_spec(fig, format="png") # serializes figure
    binary = await kaleido.to_image_w_browser(spec, browser)
    # something to write the binary to a file, async better
```
I used a blocking image write but still got 2.5X - it only uses one browser.

You can also wait for kaleido2 if you don't want to use a branch and this and it'll have a nice API, it's going to be a weekend sprint.

Sorry for the delay!

**Comment by s-banach** [comment]
@ayjayt How do I get the bytes for an image with the new API? Does `write_fig` support writing to BytesIO?

**Comment by gvwilson** [comment]
@s-banach can you please try the `write_images` function in plotly.py 6.1.2 and let us know if they resolve the issue? thanks - @gvwilson

**Comment by s-banach** [comment]
What I'm saying is that I used to do this

```python
img_bytes = pio.to_image(fig, format="svg", width=width, height=height)
```

and now I have to do the following, where `k` is an instance of `Kaleido`:

```python
    img_bytes = await k.calc_fig(
        fig,
        opts={"format": "svg", "width": width, "height": height},
    )
  ```

I'm not trying to write the images to a file.

I'm not sure whether the docstring for `calc_fig()`, claiming "this function does not support parallelism", is accurate.

**Comment by ayjayt** [comment]
Theres plans to add bytesio support, but that's a different issue so I'm going to spin your comment off into a new issue

**Comment by ayjayt** [comment]
I'm closing this as it *must be done* in kaleido v1.2.0.

Kaleido can achieve the same performance as before using 

```python
import kaleido
kaleido.start_sync_server
```

Thanks!
